pkg/apis/batch/validation/validation.go

```diff
@@ -546,6 +546,43 @@ func ValidateJobTemplateSpec(spec *batch.JobTemplateSpec, fldPath *field.Path, o
 	return allErrs
 }
 
+func validateCompletions(spec, oldSpec batch.JobSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	isIndexedJob := spec.CompletionMode != nil && *spec.CompletionMode == batch.IndexedCompletion
+	atLeastOneCompletionsNil := spec.Completions == nil || oldSpec.Completions == nil
+	bothCompletionsNil := spec.Completions == nil && oldSpec.Completions == nil
+
+	// Non-indexed Jobs cannot have completions set at all.
```

Comment:
> Non-indexed Jobs cannot have completions set at all.

I'm pretty sure that's not accurate

Reply:
I think Daniel meant "updated", not "set"

Reply:
Updated comment